protocols/mp-ecdsa/src/client.rs

```diff
@@ -0,0 +1,241 @@
+use crate::util::{find_authorities_change, DebugLogger};
+use crate::MpEcdsaProtocolConfig;
+use dkg_runtime_primitives::crypto::AuthorityId;
+use dkg_runtime_primitives::{AuthoritySet, AuthoritySetId, DKGApi, MaxAuthorities, MaxProposalLength};
+use gadget_core::gadget::substrate::Client;
+use sc_client_api::{BlockchainEvents, FinalityNotifications, HeaderBackend, ImportNotifications};
+use sp_api::{NumberFor, ProvideRuntimeApi};
+use std::error::Error;
+use std::sync::Arc;
+use tokio::sync::Mutex;
+use webb_gadget::{Backend, Block, BlockImportNotification, FinalityNotification, Header};
+use crate::client::types::AnticipatedKeygenExecutionStatus;
+use crate::keystore::DKGKeystore;
+
+pub struct MpEcdsaClient<B, BE, C> {
+    client: Arc<C>,
+    key_store: DKGKeystore,
+    finality_notification_stream: Arc<Mutex<FinalityNotifications<B>>>,
+    block_import_notification_stream: Arc<Mutex<ImportNotifications<B>>>,
+    latest_finality_notification: Arc<parking_lot::Mutex<Option<FinalityNotification<B>>>>,
+    logger: DebugLogger,
+    _block: std::marker::PhantomData<BE>,
+}
+
+pub async fn create_client<B: Block, BE: Backend<B>>(
+    _config: &MpEcdsaProtocolConfig,
+) -> Result<MpEcdsaClient<B, BE, ()>, Box<dyn Error>> {
+    Ok(MpEcdsaClient {
+        _block: std::marker::PhantomData,
+    })
+}
+
+pub trait ClientWithApi<B, BE>:
+    BlockchainEvents<B> + HeaderBackend<B> + ProvideRuntimeApi<B> + Send + Sync
+where
+    B: Block,
+    BE: Backend<B>,
+    <Self as ProvideRuntimeApi<B>>::Api:
+        DKGApi<B, AuthorityId, NumberFor<B>, MaxProposalLength, MaxAuthorities>,
+{
+}
+
+impl<B, BE, T> ClientWithApi<B, BE> for T
+where
+    B: Block,
+    BE: Backend<B>,
+    T: BlockchainEvents<B> + HeaderBackend<B> + ProvideRuntimeApi<B> + Send + Sync,
+    <T as ProvideRuntimeApi<B>>::Api:
+        DKGApi<B, AuthorityId, NumberFor<B>, MaxProposalLength, MaxAuthorities>,
+{
+}
+
+impl<B: Block, BE: Backend<B>, C: ClientWithApi<B, BE>> MpEcdsaClient<B, BE, C> {
+    pub async fn validator_set(
```

Comment:
This isn't exactly relevant anymore. The parties participating in the job are taken directly from the job.

Reply:
You can get the active validator set from the session API now or using `subxt`.